### DISCUSSION TITLE
Decrypt/Encrypt Function

### DISCUSSION THREAD

**Discussion by Kaesewuerfel** [laravel/framework#35777] [General] [answered]
How ist laravel decrypting and encrypting my strings?

**Comment by veelasky** [comment, marked as answer]
what do you mean by how?

if you refer to the usage (on how to use them), see [documentation](https://laravel.com/docs/8.x/encryption#using-the-encrypter)
```php
use Illuminate\Support\Facades\Crypt;

// encrypt string to hash
Crypt::encryptString($string);

// decrypt encrypted to the original
Crypt::decryptString($encryptedValue);
```

But if you interested to on how it works, you can always go to the source code [here](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Encryption/Encrypter.php#L82).


TLDR; Laravel use `openssl_encrypt()` and `openssl_decrypt() respectively to generate encrypted string, with the default cipher `AES-128-CBC`
